docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -129,13 +138,53 @@ For an end user, the UX to update the kubernetes version of the control plane an
 As an end user, I want to be able to easily scale up/down the number of replicas for each set of worker nodes in the cluster.
 - Currently, (for a cluster with 3 machine deployments) this is possible by updating these three different objects representing the sets of worker nodes in the pool. An easier user experience would be to update a single object to enable the scaling of multiple sets of worker nodes.
 
+#### Story 4 - Use ClusterClass to easily modify Clusters
+As an end user, I want to be able to easily change the configuration of all Clusters of a `ClusterClass`. For example, I want to be able to change the ControlPlane configuration by  
+changing the ControlPlane ref to point to another ControlPlane with a different configuration. The same should be possible for all templates referenced in a ClusterClass.
+
+**Notes**:
+- Only compatible changes (as specified in [ClusterClass compatibility](#clusterclass-compatibility)) should be allowed.
+- Changes to InfrastructureMachineTemplates and BootstrapTemplates should be rolled out according to the established operational practices documented in 
+[Updating Machine Infrastructure and Bootstrap Templates](https://cluster-api.sigs.k8s.io/tasks/updating-machine-templates.html), i.e. "template rotation".
+- There are provider-specific incompatible changes which cannot be validated in a "core" webhook, e.g. changing an immutable field of `KubeadmControlPlane`. Those changes will inevitably lead to
+  errors during topology reconciliation. Those errors should be surfaced on the Cluster resource. To mitigate this issue, we should consider implementing a clusterctl command to simulate changes like 
```

Comment:
```suggestion
  errors during topology reconciliation. Those errors should be surfaced on the Cluster resource. 
```
Maybe include the clusterctl idea in 'future work' or directly as an issue. It becomes a weird implementation thought here, not tracked and not clear.

Reply:
@fabriziopandini WDYT?

Reply:
Ah thought it's an another location. I'll just drop it. We have to revisit the clusterctl additions anyway.